test/test_extended_models.py

```diff
@@ -103,24 +103,97 @@ def test_weights_deserializable(name):
         assert pickle.loads(pickle.dumps(weights)) is weights
 
 
+def get_models_from_module(module):
+    return [
+        v.__name__
+        for k, v in module.__dict__.items()
+        if callable(v) and k[0].islower() and k[0] != "_" and k not in models._api.__all__
+    ]
+
+
 @pytest.mark.parametrize(
     "module", [models, models.detection, models.quantization, models.segmentation, models.video, models.optical_flow]
 )
 def test_list_models(module):
-    def get_models_from_module(module):
-        return [
-            v.__name__
-            for k, v in module.__dict__.items()
-            if callable(v) and k[0].islower() and k[0] != "_" and k not in models._api.__all__
-        ]
-
     a = set(get_models_from_module(module))
     b = set(x.replace("quantized_", "") for x in models.list_models(module))
 
     assert len(b) > 0
     assert a == b
 
 
+@pytest.mark.parametrize(
+    "include_filters",
+    [
+        None,
+        [],
+        (),
+        {},
+        "*resnet*",
+        ["*alexnet*"],
+        "*not-existing-model-for-test?",
+        ["*resnet*", "*alexnet*"],
+        ["*resnet*", "*alexnet*", "*not-existing-model-for-test?"],
+        ("*resnet*", "*alexnet*"),
+        set(["*resnet*", "*alexnet*"]),
+        {"*resnet*": "test", "*alexnet*": "test"},
+    ],
+)
+@pytest.mark.parametrize(
+    "exclude_filters",
+    [
+        None,
+        [],
+        (),
+        {},
+        "*resnet*",
+        ["*alexnet*"],
+        ["*not-existing-model-for-test?"],
+        ["resnet34", "*not-existing-model-for-test?"],
+        ["resnet34", "*resnet1*"],
+        ("resnet34", "*resnet1*"),
+        set(["resnet34", "*resnet1*"]),
+        {"resnet34": "test", "**resnet1**": "test"},
+    ],
+)
+def test_list_models_filters(include_filters, exclude_filters):
+    module = models
+    b = set(
+        x.replace("quantized_", "")
+        for x in models.list_models(module, include=include_filters, exclude=exclude_filters)
+    )
+    # Test include filters
+    # string filter
+    if isinstance(include_filters, str):
+        include_filters_base = include_filters.strip("*?")
+        a = set(x for x in get_models_from_module(module) if include_filters_base in x)
```

Comment:
This can probably just be

```py
    if include_filters and isinstance(include_filters, str):
        include_filters = [include_filters]
```

like in the code, and then the `elif` below should just be `if`. This avoids having the same logic duplicated in multiple places (i.e. the calls to `get_models_from_module`)